crypto/dsa/dsa_sign.c

```diff
@@ -157,6 +157,11 @@ int ossl_dsa_sign_int(int type, const unsigned char *dgst, int dlen,
 {
     DSA_SIG *s;
 
+    if (sig == NULL) {
```

Comment:
would it be safe to move this code down below the legacy case (line 165-174)?

Reply:
No, I don't think so.  Which branch is taken depends on the dsa object,
that is passed to `DSA_sign()`
only thing that is sure that both possible ways will not create a deterministic
signature, thus the correct thing to do is return the max. possible length here.

Reply:
What I am asking is, is there any way that line 172 should be run instead of the code you now run when sig is NULL

Reply:
I dont see why, the user is just asking for the memory size he needs to allocate,
and of course any error may happen when the call is repeated with sig != NULL.
This replicates the code pattern how `EVP_DigestSign` is to be used:
```C
EVP_DigestSignInit(md_ctx, NULL, EVP_sha256(), NULL, dsa);
EVP_DigestSign(md_ctx, NULL, &sig_len, tbs, tbs_len);
sig = (unsigned char*) OPENSSL_malloc(sig_len);
EVP_DigestSign(md_ctx, sig, &sig_len, tbs, tbs_len);
```
just with `DSA_sign` like this:
```C
DSA_sign(NID_sha256, dgst, dgst_len, NULL, &sig_len, dsa);
sig = (unsigned char*) OPENSSL_malloc(sig_len);
DSA_sign(NID_sha256, dgst, dgst_len, sig, &sig_len, dsa);
```